server/openejb-client/src/main/java/org/apache/openejb/client/JNDIContext.java

```diff
@@ -244,14 +243,16 @@ public Context getInitialContext(final Hashtable environment) throws NamingExcep
             providerUrl = addMissingParts(providerUrl);
             location = new URI(providerUrl);
         } catch (final URISyntaxException e) {
-            throw (ConfigurationException) new ConfigurationException("Property value for " +
-                    Context.PROVIDER_URL +
-                    " invalid: " +
-                    providerUrl +
-                    " - " +
-                    e.getMessage()).initCause(e);
+            throw (ConfigurationException) new ConfigurationException(
+                    "Property value for " + Context.PROVIDER_URL + " invalid: " + providerUrl + " - " + e.getMessage())
+                            .initCause(e);
         }
         this.server = new ServerMetaData(location);
+        String securityPrincipal = (String) env.get(Context.SECURITY_PRINCIPAL);
```

Comment:
Use different vendor properties for the Authentication header credentials, to allow the usage of different credentials for the JNDI call, or to omit the JNDI authentication.